src/mono/mono/mini/simd-intrinsics.c

```diff
@@ -2078,6 +2098,29 @@ emit_vector_2_3_4 (MonoCompile *cfg, MonoMethod *cmethod, MonoMethodSignature *f
 	case SN_CopyTo:
 		// FIXME:
 		return NULL;
+	case SN_Clamp: {
+		if (!(!fsig->hasthis && fsig->param_count == 3 && mono_metadata_type_equal (fsig->ret, type) && mono_metadata_type_equal (fsig->params [0], type) && mono_metadata_type_equal (fsig->params [1], type) && mono_metadata_type_equal (fsig->params [2], type)))
+			return NULL;
+
+		MonoInst *max = emit_simd_ins (cfg, klass, OP_XBINOP, args[0]->dreg, args[1]->dreg);
+		max->inst_c0 = OP_FMAX;
+		max->inst_c1 = MONO_TYPE_R4;
+
+		MonoInst *min = emit_simd_ins (cfg, klass, OP_XBINOP, max->dreg, args[2]->dreg);
+		min->inst_c0 = OP_FMIN;
+		min->inst_c1 = MONO_TYPE_R4;
+
+		return min;
+	}
+	case SN_Distance: {
+	case SN_DistanceSquared: {
+	case SN_Length: {
+	case SN_LengthSquared: {
+	case SN_Lerp: {
+	case SN_Normalize: {
+		// FIXME:
+		return NULL;
```

Comment:
All of these, except `Lerp`, depend on `Dot` and half also depend on `Sqrt`.

I initially tried to extract the `SN_Dot` handling to a `emit_dot_vector` helper (as there exists a `emit_sum_vector` helper). However, the general logic for `SN_Dot` is not correct for `Vector2/3/4` here and it is out of scope to ensure that is correctly handled.

If someone could give me some pointers on how to check which of `Vector2/3/4` are being processed, I can fix the logic to do the right thing in a follow up PR.

Namely, the current logic is always using `0xF1` which is "dot product all four elements and place result in element 0". However, `Vector2` should be using `0x31` and `Vector3` should be using `0x71`

Likewise, for `Normalize` we should be getting a full `SIMD` result (not a scalar) and so we want `dot` but using `0x3F`, `0x7F`, or `0xFF` respectively.

Reply:
`Sqrt` not being supported on WASM yet also makes the logic a bit more complicated than would be ideal.